ScreenControl_Web/src/InputControllers/WebInputController.ts

```diff
@@ -0,0 +1,215 @@
+            // the cursor
+            if (this.lastHoveredElement !== null) {
+                let outEvent: PointerEvent = new PointerEvent("pointerout", this.activeEventProps);
+                this.lastHoveredElement.dispatchEvent(outEvent);
+            }
+
+            if (_element !== null) {
+                let overEvent: PointerEvent = new PointerEvent("pointerover", this.activeEventProps);
+                _element.dispatchEvent(overEvent);
+            }
+
+            if (this.enterLeaveEnabled) {
+                this.HandleEnterLeaveBehaviour(_element);
+            }
+        }
+
+        let moveEvent: PointerEvent = new PointerEvent("pointermove", this.activeEventProps);
+        _element?.dispatchEvent(moveEvent);
+
+        this.lastHoveredElement = _element;
+    }
+
+    // Function: HandleInputAction
+    // Called with each <ClientInputAction> as it comes into the <ServiceConnection>. Updates the
+    // underlying InputModule and EventSystem based on the incoming actions.
+    //
+    // Parameters:
+    //     _inputData - The latest Action to arrive via the <ServiceConnection>.
+    protected HandleInputAction(_inputData: ClientInputAction): void {
+        _inputData.CursorPosition[1] = window.innerHeight - _inputData.CursorPosition[1];
+        super.HandleInputAction(_inputData);
+
+        let elementAtPos: Element | null = this.GetTopNonCursorElement(_inputData.CursorPosition);
+
+        this.activeEventProps.clientX = _inputData.CursorPosition[0];
+        this.activeEventProps.clientY = _inputData.CursorPosition[1];
+
+        switch (_inputData.InputType) {
+            case InputType.CANCEL:
+                let cancelEvent: PointerEvent = new PointerEvent("cancel", this.activeEventProps);
+                if (elementAtPos != null) {
+                    let parentTree = this.GetOrderedParents(elementAtPos);
+
+                    parentTree.forEach((parent: Node | null) => {
+                        if (parent != null) {
+                            parent.dispatchEvent(cancelEvent);
+                        }
+                    });
+                }
+                break;
+
+            case InputType.MOVE:
+                this.HandleMove(elementAtPos);
+                break;
+
+            case InputType.DOWN:
+                let downEvent: PointerEvent = new PointerEvent("pointerdown", this.activeEventProps);
+                this.DispatchToTarget(downEvent, elementAtPos);
+                break;
+
+            case InputType.UP:
+                let upEvent: PointerEvent = new PointerEvent("pointerdown", this.activeEventProps);
+                this.DispatchToTarget(upEvent, elementAtPos);
+                break;
+        }
+    }
+
+    // Gets the stack of elements (topmost->bottommost) at this position and return the first non-
+    // cursor element. Depends on all cursor elements being branded with the "cursor" class.
```

Comment:
This suggests the documentation for the cursors should be explicit about needing the class of "cursor"